core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorActor.scala

```diff
@@ -208,9 +208,14 @@ trait SubmissionMonitor extends FutureSupport with LazyLogging {
     //query Cromwell again, and restart from the top of this function with the results.
     workflowsWithOutputsF flatMap { workflowsWithOutputs =>
       datasource.inTransaction { dataAccess =>
-        handleOutputs(workflowsWithOutputs, dataAccess) flatMap { _ =>
-          checkOverallStatus(dataAccess) map {
-            shouldStop => StatusCheckComplete(shouldStop)
+        //re-fetch the workflow records because their version number has been bumped by the batchUpdateStatus
+        DBIO.sequence(workflowsWithOutputs map { case (rec, outputs) =>
+          dataAccess.workflowQuery.findWorkflowById(rec.id).result map { recs => (recs.head, outputs) }
```

Comment:
I'm worried about this usage of `workflowQuery.findWorkflowById`. It looks like it makes N database queries, where N is the number of workflows with outputs.

At minimum, use `workflowQuery.findWorkflowByIds` instead so you're only making one db query.

I'm also not seeing the need to re-fetch the workflow records. I'm sure you're correct that the `workflowsWithOutputs` variable you have at this point is out-of-date, but I'm not seeing, in `handleOutputs` or `checkOverallStatus`, where this matters.